File: easy/0520_detect_capital/solution.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        if len(word) == 1:
            return True

        is_upper = word[0].isupper() and word[1].isupper()

        for i in range(1, len(word)):
            if word[i].islower() == is_upper:
                return False

        return True

File: easy/0520_detect_capital/solution.cpp
#include <string>

class Solution {
public:
  bool detectCapitalUse(std::string word) {
    if (word.size() == 1)
      return true;

    if (isupper(word[0]) && isupper(word[1])) {
      for (size_t i = 1; i < word.size(); ++i) {
        if (islower(word[i]))
          return false;
      }
    } else {
      for (size_t i = 1; i < word.size(); ++i) {
        if (isupper(word[i]))
          return false;
      }
    }

    return true;
  }
};
